### PR TITLE
#43 - Add interfaces for framework models of Shopware 6

### PR DESCRIPTION
**Pull request by rmakara**
**Changes:**

* Add interfaces for `framework` group based on Shopware 6 data structures.
* These models seem to be required for covering Shopware API endpoints that we will need.
* It creates the following structure below `framework`:
  * data-abstraction-layer
  * language
  * plugin
  * pricing
  * rule
  * snippet
  * struct

**Issue:**

#43

### CODE REVIEW

File: packages/shopware-6-client/src/interfaces/models/framework/plugin/PluginTranslation.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Plugin } from "./Plugin";
+import { CustomField } from "../../common/CustomField";
+export interface PluginTranslation {
+  pluginId: string;
+  label: string | null;
+  description: string | null;
+  manufacturerLink: string | null;
+  supportLink: string | null;
+  changelog: [] | null;
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/models/framework/plugin/PluginTranslation.ts]: missing type

File: packages/shopware-6-client/src/interfaces/models/framework/rule/Rule.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { ShippingMethod } from "../../checkout/shipping/ShippingMethod";
+import { RuleCondition } from "../../content/rule/RuleCondition";
+import { CustomField } from "../../common/CustomField";
+
+export interface Rule {
+  name: string;
+  description: string | null;
+  priority: number;
+  payload: string | Rule | null;
+  moduleTypes: [] | null;
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/models/framework/rule/Rule.ts]: missing type

File: packages/shopware-6-client/src/interfaces/models/framework/language/Language.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  salesChannelTranslations: SalesChannelTranslation[] | null;
+  salesChannelTypeTranslations: SalesChannelTypeTranslation[] | null;
+  salutationTranslations: SalutationTranslation[] | null;
+  salesChannelDomains: SalesChannelDomain[] | null;
+  pluginTranslations: PluginTranslation[] | null;
+  productStreamTranslations: ProductStreamTranslation[] | null;
+  stateMachineTranslations: [] | null;
+  stateMachineStateTranslations: [] | null;
+  cmsPageTranslation: [] | null;
+  cmsSlotTranslations: [] | null;
```
[the remark this answers is not in the document]
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/framework/language/Language.ts]: Ouch, that's an use case for `Collection` interface, that I deleted in the other PR. Adding back Collection interface with simple JSDoc. 

Collection interface is being used for not many fields in Shopware and it contains `elements[]` only.

File: packages/shopware-6-client/src/interfaces/models/framework/plugin/Plugin.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  upgradeVersion: string | null;
+  installedAt: Date;
+  upgradedAt: Date;
+  iconRaw: string | null;
+  icon: string | null;
+  label: string;
+  description: string | null;
+  manufacturerLink: string | null;
+  supportLink: string | null;
+  changelog: [] | null;
```
[the remark this answers is not in the document]
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/framework/plugin/Plugin.ts]: Just an array in Shopware.

File: packages/shopware-6-client/src/interfaces/models/framework/plugin/Plugin.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  translations: PluginTranslation[] | null;
+  paymentMethods: PaymentMethod[] | null;
+  customFields: CustomField[];
+  autoload: [];
```
[the remark this answers is not in the document]
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/framework/plugin/Plugin.ts]: Just an array in Shopware.

File: packages/shopware-6-client/src/interfaces/models/framework/plugin/PluginTranslation.ts (hunk 1 above)
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/framework/plugin/PluginTranslation.ts]: Just an array in Shopware.

File: packages/shopware-6-client/src/interfaces/models/framework/rule/Rule.ts (hunk 2 above)
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/framework/rule/Rule.ts]: Just an array in Shopware.

**Pull request merged**